docs/src/getting-started/temporary_workarounds.md

```diff
@@ -62,3 +62,7 @@ Serializing arbitrary structures can be accomplished manually by composition of
 ## Optimizer
 
 The optimizing pass of the compiler is not yet implemented, therefore bytecode will be more expensive and larger than it would be in production. Note that eventually the optimizer will support zero-cost abstractions, avoiding the need for developers to go down to inline assembly to produce optimal code.
+
+## Formatter
+
+Currently, we need to parse the sway code before formatting it, and the best way to do this is to compile the code. Hence, **the formatter cannot work on sway code that does not compile**. This requirement may be changed in the future. 
```

Comment:
This might be incorrect (which is my bad). It could be that code that parses but doesn't compile can still be formatted. Please confirm @sezna.

```suggestion
Currently, we need to parse the Sway code before formatting it, and the best way to do this is to compile the code. Hence, **the formatter cannot work on Sway code that does not compile**. This requirement may be changed in the future. 
```

Reply:
In Forc’s current state we do run a full compile to format. I’m unsure why — it should hypothetically be possible after parsing, but I haven’t looked into it as I am unfamiliar with the formatter code. Ideally we should be able to just parse it.